l10n_it_asset_management/models/asset_depreciation_line.py

```diff
@@ -0,0 +1,490 @@
+    _order = "date asc, name asc"
+
+    amount = fields.Monetary()
+
+    asset_accounting_info_ids = fields.One2many(
+        "asset.accounting.info", "dep_line_id", string="Accounting Info"
+    )
+
+    asset_id = fields.Many2one(
+        "asset.asset",
+        readonly=True,
+        related="depreciation_id.asset_id",
+        store=True,
+        string="Asset",
+    )
+
+    balance = fields.Monetary(compute="_compute_balance", store=True)
+
+    base = fields.Float()
+
+    company_id = fields.Many2one(
+        "res.company",
+        readonly=True,
+        related="depreciation_id.company_id",
+    )
+
+    currency_id = fields.Many2one(
+        "res.currency",
+        readonly=True,
+        related="depreciation_id.currency_id",
+    )
+
+    date = fields.Date(required=True)
+
+    depreciation_id = fields.Many2one(
+        "asset.depreciation",
+        ondelete="cascade",
+        readonly=True,
+        required=True,
+        string="Asset depreciation",
+    )
+
+    depreciation_line_type_id = fields.Many2one(
+        "asset.depreciation.line.type", string="Depreciation Type"
+    )
+
+    depreciation_nr = fields.Integer(
+        string="Dep. Num",
+    )
+
+    depreciation_type_id = fields.Many2one(
+        "asset.depreciation.type",
+        readonly=True,
+        related="depreciation_id.type_id",
+        store=True,
+        string="Asset depreciation type",
+    )
+
+    force_dep_nr = fields.Boolean(
+        readonly=True,
+        related="depreciation_id.force_all_dep_nr",
+        string="Force Dep. Num",
+    )
+
+    move_id = fields.Many2one("account.move", string="Move")
+
+    move_type = fields.Selection(
+        [
+            ("depreciated", "Depreciation"),
+            ("historical", "Historical"),
+            ("in", "In"),
+            ("out", "Out"),
+            ("loss", "Capital Loss"),
+            ("gain", "Capital Gain"),
+        ],
+        string="Type",
+        required=True,
+    )
+
+    name = fields.Char(required=True)
+
+    partial_dismissal = fields.Boolean()
+
+    percentage = fields.Float(string="%")
+
+    requires_account_move = fields.Boolean(
+        readonly=True,
+        related="depreciation_id.type_id.requires_account_move",
+        string="Required Account Move",
+    )
+
+    requires_depreciation_nr = fields.Boolean(
+        compute="_compute_requires_depreciation_nr",
+        search="_search_requires_depreciation_nr_lines",
+        string="Requires Dep Num",
+    )
+
+    # Non-default parameter: set which `move_types` require numeration
+    _numbered_move_types = ("depreciated", "historical")
+    # Non-default parameter: set which `move_types` do not concur to
+    # asset.depreciation's `amount_residual` field compute
+    _non_residual_move_types = ("gain",)
+    # Non-default parameter: set which `move_types` get to update the
+    # depreciable amount
+    _update_move_types = ("in", "out")
+
+    @api.model
+    def create(self, vals):
```

Comment:
Dovresti usare il decoratore `@api.model_create_multi` e modificare la funzione per warning `is not overriding the create method in batch`

Reply:
Fatto